Implemented critical tests 1

lesiavl opened the pull request:
To run, do shell command in main project:
```
./bin/openprocurement_tests -A robot_tests_arguments/below_simple_basic.txt --critical level1
```

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/openprocurement/robot_tests.broker.publicportal/4)
<!-- Reviewable:end -->

mykhaly commented:
Команда запуску для нас нічого не каже, бо `robot_tests_arguments/below_simple_basic.txt` Ви нам не надіслали.
Підозрюю, що команди запуску мають бути отакими:

```
bin/op_tests -s openProcedure -A robot_tests_arguments/below_simple.txt -v role:viewer -v broker:PublicPortal --critical level1
bin/op_tests -s openProcedure -A robot_tests_arguments/openua_simple.txt -v role:viewer -v broker:PublicPortal --critical level1
bin/op_tests -s openProcedure -A robot_tests_arguments/openeu_simple.txt -v role:viewer -v broker:PublicPortal --critical level1
bin/op_tests -s reporting -v role:viewer -v broker:PublicPortal --critical level1
bin/op_tests -s negotiation -v role:viewer -v broker:PublicPortal --critical level1
bin/op_tests -s negotiation.quick -v role:viewer -v broker:PublicPortal --critical level1
```

Чи я помиляюся?

---

Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 18 unresolved discussions.

---

_[publicportal.py, line 1 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaQ6CSph_9WK_QueJ2:-KNaQ6CTekFKkW19mKCI:162921541) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.py#L1)):_

> ``` Python
> def create_xpath(xpath, variable):
> ```

Що Ви передаєте тут у якості аргументу `xpath`?

---

_[publicportal.py, line 6 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaNknb4mTfZWOhimmp:-KNaNknb4mTfZWOhimmq:-1861209150) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.py#L6)):_

> ``` Python
> 
> def prune_amount(str):
>     return round(int(''.join([s for s in str.split() if s.isdigit()])))
> ```

З якою точністю тут відбувається округлення?
Аргумент з назвою `str` варто замінити на `string`, бо `str` - одна з вбудованих функцій.
Я так розумію, що число для Вас приходить у форматі `123 456 789.01`?
Якщо так, то на Вашому місці я б замінив цей код на отакий:

```
def prune_amount(string):
    return round(float(string.replace(' ', '')), 2)
```

---

_[publicportal.robot, line 12 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaQHsY8RCU-14hrxAc:-KNaQHsY8RCU-14hrxAd:129500837) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.robot#L12)):_

> ``` RobotFramework
> 
> ${settings.global.timeout}  10
> ${locator.image.verification}     xpath=/html/body/div/nav[2]/div/div[1]/a[1]/img
> ```

А чи не хардкод це? І всюди далі теж.
Якщо в розмітці зміниться щось, то цей код, ймовірно, зламається.

---

_[publicportal.robot, line 19 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaQ_Z0SwPXrSn39a69:-KNaQ_Z0SwPXrSn39a6A:1083930954) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.robot#L19)):_

> ``` RobotFramework
> ${locator.go.to.auction}                xpath=//*[@href="/tender/{}/"]
> ${locator.auction.get.link}             xpath=//*[@id="sticky-wrapper"]/div/div/a
> # Button  ${locator.auction.get.link} doesn't appear with tender e.g. UA-2016-07-20-000247-1, but with e.g. UA-2016-07-21-000035-1 it's OK
> ```

Що це?

---

_[publicportal.robot, line 29 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaQhK13aEY0XycEeIN:-KNaQhK13aEY0XycEeIO:2060155343) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.robot#L29)):_

> ``` RobotFramework
> 
> *** Keywords ***
> ```

Тут не потрібно порожнього рядка.

---

_[publicportal.robot, line 31 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaQkZuU-XI1t86L-g9:-KNaQkZuU-XI1t86L-gA:741967654) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.robot#L31)):_

> ``` RobotFramework
> 
> Підготувати клієнт для користувача
>   Set Global Variable  ${PUBLICPORTAL_MODIFICATION_DATE}  ${EMPTY}
> ```

Секцію `[Arguments]` і `[Documentation]` варто поставити перед тілом ключового слова.

---

_[publicportal.robot, line 39 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaQvQyOiw9mL3AVxaQ:-KNaQvQyOiw9mL3AVxaR:150291119) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.robot#L39)):_

> ``` RobotFramework
>   Set Window Position  @{users.users['${username}'].position}
> 
> #Basic Test: opens browser, goes to the home page, resizes the window (users.yaml)
> ```

Це коментар до `Підготувати клієнт для користувача`? Якщо так - запхайте в секцію `[Documentation]`.

---

_[publicportal.robot, line 44 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaR7ZpG4aZB1hu4h8J:-KNaR7ZpG4aZB1hu4h8K:-101675746) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.robot#L44)):_

> ``` RobotFramework
> Пошук тендера по ідентифікатору
>   [Arguments]  ${username}  ${tender_uaid}
>   Reload Page
> ```

Навіщо перезавантажувати сторінку перед тим, як йти на якусь іншу?

---

_[publicportal.robot, line 55 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaROoMC6ukhY2hyjdY:-KNaROoMC6ukhY2hyjdZ:-716780844) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.robot#L55)):_

> ``` RobotFramework
>   wait until page contains element  ${locator.auction.get.link}  ${settings.global.timeout}
>   ${proceed_url}=  Get Element Attribute  ${locator.auction.get.link}@href
>   [return]  ${proceed_url}
> ```

Мені здається, чи це ключове слово не робить нічого корисного, крім виклику `Пошук тендера по ідентифікатору`? Якщо так - заберіть все лишнє.

---

_[publicportal.robot, line 64 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaR_QR1qIxQqIQiR--:-KNaR_QR1qIxQqIQiR-0:1084040507) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.robot#L64)):_

> ``` RobotFramework
>   ${year}=  Get Substring  ${date}   5
>   ${return_value}=  Convert To String  ${month}${day}${year}
>   [return]  ${return_value}
> ```

Що це за квадратик після `${tender_uaid}`?

---

_[publicportal.robot, line 66 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaRezMpd7i-JRMLFCN:-KNaRezMpd7i-JRMLFCO:-944024027) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.robot#L66)):_

> ``` RobotFramework
>   [return]  ${return_value}
> ```

Дивна назва для змінної...
Раджу перейменувати на `${return_value}`.

---

_[publicportal.robot, line 100 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaS5iBUttSQBWU2WWh:-KNaS5iBUttSQBWU2WWi:1791471188) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.robot#L100)):_
Зайві логи тут і нижче. Локально можете таке логувати під час розробки, але в репозиторії це непотрібне.

---

_[publicportal.robot, line 110 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaSI7_ZG2CXD4RtRwB:-KNaSI7_ZG2CXD4RtRwC:-177359294) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.robot#L110)):_
Видаліть цей рядок, якщо він не потрібен.

---

_[publicportal.robot, line 111 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaSPXwHPjDa3IVONFz:-KNaSPXwHPjDa3IVONG-:-145168129) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.robot#L111)):_
Тут варто додати назву файлу (`publicportal`), з якого брати `Пошук тендера по ідентифікатору`.

---

_[publicportal.robot, line 112 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaSMSi_gRr-QQElk9y:-KNaSMSjlP22nayGQXAV:2082920413) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.robot#L112)):_
Це вже логується в `Пошук тендера по ідентифікатору`.

---

_[publicportal.robot, line 114 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaSZVcoU9Gkz-uA-U5:-KNaSZVcoU9Gkz-uA-U6:-1655373504) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.robot#L114)):_
Ключові слова, вбудовані в robot framework, чи з його бібліотек, варто писати з великої букви. Бажано - кожне слово з великої букви, через пробіл.

---

_[publicportal.robot, line 116 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaSqApL7GWPR9yNi_q:-KNaSqApL7GWPR9yNi_r:-417764468) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.robot#L116)):_
Те ж, що й вище.

---

_[publicportal.robot, line 121 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaStku-Gh1y8vBZysR:-KNaStku-Gh1y8vBZysS:1825043985) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.robot#L121)):_
Я думаю, що краще це ключове слово написати на пітоні. Гарніше і природніше буде виглядати. Та й назвати варто якось типу `parse_date_publicportal`.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-:-KNaMl4yFQNI2nGmkBXV:-1502929949)_

<!-- Sent from Reviewable.io -->

lesiavl commented:
Review status: 0 of 2 files reviewed at latest revision, 18 unresolved discussions.

---

_[publicportal.py, line 1 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaQ6CSph_9WK_QueJ2:-KNbsmG-jdVYc23b-Hmb:64126408) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.py#L1)):_
<details><summary>

<i>Previously, mykhaly (Yurii Mykhalchuk) wrote…</i></summary>



> Що Ви передаєте тут у якості аргументу `xpath`?
> </details>
> змінила на (xpath, tender_uaid) так як його треба вставляти в локатор

---

_[publicportal.py, line 6 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaNknb4mTfZWOhimmp:-KNbtk0AfLC4mhLKQWH9:1280826138) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.py#L6)):_
<details><summary>

<i>Previously, mykhaly (Yurii Mykhalchuk) wrote…</i></summary>



> З якою точністю тут відбувається округлення?
> Аргумент з назвою `str` варто замінити на `string`, бо `str` - одна з вбудованих функцій.
> Я так розумію, що число для Вас приходить у форматі `123 456 789.01`?
> Якщо так, то на Вашому місці я б замінив цей код на отакий:
> 
> ```
> def prune_amount(string):
>     return round(float(string.replace(' ', '')), 2)
> ```
> 
> </details>
> змінила на даний варіант

---

_[publicportal.robot, line 12 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaQHsY8RCU-14hrxAc:-KNbtncD1mhW8rAGJnpX:-1575425737) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.robot#L12)):_
<details><summary>

<i>Previously, mykhaly (Yurii Mykhalchuk) wrote…</i></summary>



> А чи не хардкод це? І всюди далі теж.
> Якщо в розмітці зміниться щось, то цей код, ймовірно, зламається.
> </details>
> змінила на більш гнучкий варіант

---

_[publicportal.robot, line 19 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaQ_Z0SwPXrSn39a69:-KNbtsOHn7iWxAiPqrAX:-444133959) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.robot#L19)):_
<details><summary>

<i>Previously, mykhaly (Yurii Mykhalchuk) wrote…</i></summary>



> Що це?
> </details>
> видалила

---

_[publicportal.robot, line 29 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaQhK13aEY0XycEeIN:-KNbttgGzriB_ysjEr5W:166854) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.robot#L29)):_
<details><summary>

<i>Previously, mykhaly (Yurii Mykhalchuk) wrote…</i></summary>



> Тут не потрібно порожнього рядка.
> </details>
> +

---

_[publicportal.robot, line 31 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaQkZuU-XI1t86L-g9:-KNbtuzhhtdiw8xHoWaA:166854) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.robot#L31)):_
<details><summary>

<i>Previously, mykhaly (Yurii Mykhalchuk) wrote…</i></summary>



> Секцію `[Arguments]` і `[Documentation]` варто поставити перед тілом ключового слова.
> </details>
> +

---

_[publicportal.robot, line 39 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaQvQyOiw9mL3AVxaQ:-KNbtwHJSnJk9QpPRd-b:-444133959) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.robot#L39)):_
<details><summary>

<i>Previously, mykhaly (Yurii Mykhalchuk) wrote…</i></summary>



> Це коментар до `Підготувати клієнт для користувача`? Якщо так - запхайте в секцію `[Documentation]`.
> </details>
> видалила

---

_[publicportal.robot, line 44 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaR7ZpG4aZB1hu4h8J:-KNbu1Y2CMe9HJjNhkZe:-279664356) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.robot#L44)):_
<details><summary>

<i>Previously, mykhaly (Yurii Mykhalchuk) wrote…</i></summary>



> Навіщо перезавантажувати сторінку перед тим, як йти на якусь іншу?
> </details>
> змінила

---

_[publicportal.robot, line 66 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaRezMpd7i-JRMLFCN:-KNbuIXfEuJi6DuYqLNx:890219483) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.robot#L66)):_
<details><summary>

<i>Previously, mykhaly (Yurii Mykhalchuk) wrote…</i></summary>



> Дивна назва для змінної...
> Раджу перейменувати на `${return_value}`.
> </details>
> перейменувала

---

_[publicportal.robot, line 100 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaS5iBUttSQBWU2WWh:-KNbuKx0UgW70krvizvv:-444133959) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.robot#L100)):_
<details><summary>

<i>Previously, mykhaly (Yurii Mykhalchuk) wrote…</i></summary>



> Зайві логи тут і нижче. Локально можете таке логувати під час розробки, але в репозиторії це непотрібне.
> </details>
> видалила

---

_[publicportal.robot, line 110 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaSI7_ZG2CXD4RtRwB:-KNbuMghs9VcR-Zg3TOI:-444133959) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.robot#L110)):_
<details><summary>

<i>Previously, mykhaly (Yurii Mykhalchuk) wrote…</i></summary>



> Видаліть цей рядок, якщо він не потрібен.
> </details>
> видалила

---

_[publicportal.robot, line 112 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaSMSi_gRr-QQElk9y:-KNbuUQyFySWrv4wzS9D:1856730152) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.robot#L112)):_
<details><summary>

<i>Previously, mykhaly (Yurii Mykhalchuk) wrote…</i></summary>



> Це вже логується в `Пошук тендера по ідентифікатору`.
> </details>
> видалила всі логи

---

_[publicportal.robot, line 114 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaSZVcoU9Gkz-uA-U5:-KNbuXSFeedG8suM26f-:1374806953) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.robot#L114)):_
<details><summary>

<i>Previously, mykhaly (Yurii Mykhalchuk) wrote…</i></summary>



> Ключові слова, вбудовані в robot framework, чи з його бібліотек, варто писати з великої букви. Бажано - кожне слово з великої букви, через пробіл.
> </details>
> виправлено

---

_[publicportal.robot, line 121 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaStku-Gh1y8vBZysR:-KNbu_EpVAkInsoT2EFM:-653972615) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.robot#L121)):_
<details><summary>

<i>Previously, mykhaly (Yurii Mykhalchuk) wrote…</i></summary>



> Я думаю, що краще це ключове слово написати на пітоні. Гарніше і природніше буде виглядати. Та й назвати варто якось типу `parse_date_publicportal`.
> </details>
> додала ключове словл у файл publicportal.py

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4)_

<!-- Sent from Reviewable.io -->

lesiavl commented:
Review status: 0 of 2 files reviewed at latest revision, 6 unresolved discussions.

---

_[publicportal.robot, line 55 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaROoMC6ukhY2hyjdY:-KNc0pJOTKjogAPCrTeZ:166854) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.robot#L55)):_
<details><summary>

<i>Previously, mykhaly (Yurii Mykhalchuk) wrote…</i></summary>



> Мені здається, чи це ключове слово не робить нічого корисного, крім виклику `Пошук тендера по ідентифікатору`? Якщо так - заберіть все лишнє.
> </details>
> +

---

_[publicportal.robot, line 64 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaR_QR1qIxQqIQiR--:-KNc0llMGCOZdlaFH0jJ:-2081635383) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.robot#L64)):_
<details><summary>

<i>Previously, mykhaly (Yurii Mykhalchuk) wrote…</i></summary>



> Що це за квадратик після `${tender_uaid}`?
> </details>
> ніби виправила

---

_[publicportal.robot, line 111 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaSPXwHPjDa3IVONFz:-KNc0jSh93rEwzLhw_I2:166854) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.robot#L111)):_
<details><summary>

<i>Previously, mykhaly (Yurii Mykhalchuk) wrote…</i></summary>



> Тут варто додати назву файлу (`publicportal`), з якого брати `Пошук тендера по ідентифікатору`.
> </details>
> +

---

_[publicportal.robot, line 116 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaSqApL7GWPR9yNi_q:-KNbvYXBZ-XjZMxc-tFs:166854) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.robot#L116)):_
<details><summary>

<i>Previously, mykhaly (Yurii Mykhalchuk) wrote…</i></summary>



> Те ж, що й вище.
> </details>
> +

---

_[publicportal.robot, line 95 [r2]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNbycqhhUmI1W5KCTIH:-KNbycqhhUmI1W5KCTII:455455258) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/4035447fbe0d30c2f1d8105e34cbb3cf483edfd2/publicportal.robot#L95)):_

> ача
>   [Arguments]  ${username}  ${tender_uaid}  ${url}
>   Пошук тендера по ідентифікатору  ${username}  ${tender_uaid}

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4)_

<!-- Sent from Reviewable.io -->

mykhaly commented:
Reviewed 1 of 1 files at r3.
Review status: 1 of 2 files reviewed at latest revision, 8 unresolved discussions.

---

_[publicportal.py, line 1 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaQ6CSph_9WK_QueJ2:-KNbzYblADZPkBG9frTx:1906224205) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.py#L1)):_
<details><summary>

<i>Previously, lesiavl (Lesia Velychko) wrote…</i></summary>



> змінила на (xpath, tender_uaid) так як його треба вставляти в локатор
> 
> </details>
> Я не про те питався.

---

_[publicportal.robot, line 55 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaROoMC6ukhY2hyjdY:-KNc7BpUQNOS9kPfiEAN:-1463955547) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.robot#L55)):_
<details><summary>

<i>Previously, lesiavl (Lesia Velychko) wrote…</i></summary>



> +
> </details>
> Не забрали все лишнє -_-

---

_[publicportal.robot, line 64 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaR_QR1qIxQqIQiR--:-KNc7JWqtxz172I8uA3j:-1646203720) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.robot#L64)):_
<details><summary>

<i>Previously, lesiavl (Lesia Velychko) wrote…</i></summary>



> ніби виправила
> </details>
> No.
> Хай вже буде.

---

_[publicportal.robot, line 142 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNc7ubWIdZiMOSTYP9D:-KNc7ubWIdZiMOSTYP9E:-1777461687) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.robot#L142)):_
Чому це ключове слово видалили, а інші, в секції `NOT IMPLEMENTED KEYWORDS` - ні?

---

_[publicportal.robot, line 91 [r2]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNbwIGBaBovRf4GPhwW:-KNbwIGBaBovRf4GPhwX:-661060673) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/4035447fbe0d30c2f1d8105e34cbb3cf483edfd2/publicportal.robot#L91)):_
Здається, тут не потрібно `Convert To Number`.

---

_[publicportal.robot, line 95 [r2]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNbycqhhUmI1W5KCTIH:-KNc76b0Gctf7V23u2Wk:677187898) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/4035447fbe0d30c2f1d8105e34cbb3cf483edfd2/publicportal.robot#L95)):_
<details><summary>

<i>Previously, lesiavl (Lesia Velychko) wrote…</i></summary>



> > ача
> >   [Arguments]  ${username}  ${tender_uaid}  ${url}
> >   Пошук тендера по ідентифікатору  ${username}  ${tender_uaid}
> 
> </details>
> Wat?

---

_[publicportal.robot, line 29 [r3]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNc7T7BxdUP47_q0WQC:-KNc7T7BxdUP47_q0WQD:579854924) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/24aee6afe933385cf0dd05548d871c9b0c37873b/publicportal.robot#L29)):_

> ``` RobotFramework
>   [Arguments]  ${username}
>   [Documentation]  Відкрити браузер, створити об’єкт api wrapper, тощо
>   Set Global Variable  ${PUBLICPORTAL_MODIFICATION_DATE}  ${EMPTY}
> ```

Не бачу використання цієї змінної. Варто видалити.

---

_[publicportal.robot, line 49 [r3]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNc7cvFEZKbgzHNF4y_:-KNc7cvFEZKbgzHNF4ya:-735490278) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/24aee6afe933385cf0dd05548d871c9b0c37873b/publicportal.robot#L49)):_

> ``` RobotFramework
>   Click Element  ${xpath}
>   Wait Until Page Contains Element  ${locator.auction.get.link}  ${settings.global.timeout}
>   ${proceed_url}=  Get Element Attribute  ${locator.auction.get.link}@href
> ```

`Selenium2Library.Switch Browser` чимось відрізняється від `Switch Browser`?

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4)_

<!-- Sent from Reviewable.io -->

lesiavl commented:
Review status: 0 of 2 files reviewed at latest revision, 8 unresolved discussions.

---

_[publicportal.py, line 1 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaQ6CSph_9WK_QueJ2:-KNg_NkB_6GARmALrAJI:-499146009) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.py#L1)):_
<details><summary>

<i>Previously, mykhaly (Yurii Mykhalchuk) wrote…</i></summary>



> Я не про те питався.
> </details>
> Done.

---

_[publicportal.robot, line 55 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaROoMC6ukhY2hyjdY:-KNgRfxqgcKv_j6FSZao:-499146009) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.robot#L55)):_
<details><summary>

<i>Previously, mykhaly (Yurii Mykhalchuk) wrote…</i></summary>



> Не забрали все лишнє -_-
> </details>
> Done.

---

_[publicportal.robot, line 64 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaR_QR1qIxQqIQiR--:-KNgHTZklSWBCFpRoQjx:-499146009) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.robot#L64)):_
<details><summary>

<i>Previously, mykhaly (Yurii Mykhalchuk) wrote…</i></summary>



> No.
> Хай вже буде.
> </details>
> Done.

---

_[publicportal.robot, line 142 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNc7ubWIdZiMOSTYP9D:-KNgIsI9XcWkjBZ4N_6q:-1040976545) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.robot#L142)):_
<details><summary>

<i>Previously, mykhaly (Yurii Mykhalchuk) wrote…</i></summary>



> Чому це ключове слово видалили, а інші, в секції `NOT IMPLEMENTED KEYWORDS` - ні?
> </details>
> видалила секцію

---

_[publicportal.robot, line 91 [r2]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNbwIGBaBovRf4GPhwW:-KNgPY5LsYajeDp4dCEj:-499146009) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/4035447fbe0d30c2f1d8105e34cbb3cf483edfd2/publicportal.robot#L91)):_
<details><summary>

<i>Previously, mykhaly (Yurii Mykhalchuk) wrote…</i></summary>



> Здається, тут не потрібно `Convert To Number`.
> </details>
> Done.

---

_[publicportal.robot, line 95 [r2]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNbycqhhUmI1W5KCTIH:-KNgOIxsFnqyqkuac19d:-499146009) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/4035447fbe0d30c2f1d8105e34cbb3cf483edfd2/publicportal.robot#L95)):_
<details><summary>

<i>Previously, mykhaly (Yurii Mykhalchuk) wrote…</i></summary>



> Wat?
> </details>
> Done.

---

_[publicportal.robot, line 29 [r3]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNc7T7BxdUP47_q0WQC:-KNgRm6lDI7ao0Lh2Plm:-499146009) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/24aee6afe933385cf0dd05548d871c9b0c37873b/publicportal.robot#L29)):_
<details><summary>

<i>Previously, mykhaly (Yurii Mykhalchuk) wrote…</i></summary>



> Не бачу використання цієї змінної. Варто видалити.
> </details>
> Done.

---

_[publicportal.robot, line 49 [r3]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNc7cvFEZKbgzHNF4y_:-KNgPyjeVSN4YnscGAje:-499146009) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/24aee6afe933385cf0dd05548d871c9b0c37873b/publicportal.robot#L49)):_
<details><summary>

<i>Previously, mykhaly (Yurii Mykhalchuk) wrote…</i></summary>



> `Selenium2Library.Switch Browser` чимось відрізняється від `Switch Browser`?
> </details>
> Done.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4)_

<!-- Sent from Reviewable.io -->

mykhaly commented:
Reviewed 1 of 2 files at r4, 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

_[publicportal.py, line 1 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaQ6CSph_9WK_QueJ2:-KNgi6y-ontAhY19Crbe:-510466975) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.py#L1)):_
<details><summary>

<i>Previously, lesiavl (Lesia Velychko) wrote…</i></summary>



> Done.
> </details>
> :+1: 

---

_[publicportal.robot, line 64 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaR_QR1qIxQqIQiR--:-KNgi_NToas3fLHqCSTj:676269983) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.robot#L64)):_
<details><summary>

<i>Previously, lesiavl (Lesia Velychko) wrote…</i></summary>



> Done.
> </details>
> :+1:

---

_[publicportal.robot, line 41 [r5]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNgiIZUG6t1IfDG1sFz:-KNgiIZUG6t1IfDG1sG-:-287145709) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/053d59691ea57bbab4e2de7113f87a240c1ca29a/publicportal.robot#L41)):_

> ``` RobotFramework
>   Click Button  ${locator.tender.number}
>   Input Text    ${locator.tender.search}  ${tender_uaid}
>   ${xpath}=  insert_tender_id_into_xpath ${locator.tender.title.verification}  ${tender_uaid}
> ```

Якщо забрати цей пробіл, то виникне помилка, оскільки Robot Framework буде вважати `insert_tender_id_into_xpath ${locator.tender.title.verification}` за ключове слово, а не ключове слово і аргумент.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4)_

<!-- Sent from Reviewable.io -->

lesiavl commented:
Review status: 1 of 2 files reviewed at latest revision, 3 unresolved discussions.

---

_[publicportal.robot, line 41 [r5]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNgiIZUG6t1IfDG1sFz:-KNgvEdEZeNRyBZ3U2_k:-499146009) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/053d59691ea57bbab4e2de7113f87a240c1ca29a/publicportal.robot#L41)):_
<details><summary>

<i>Previously, mykhaly (Yurii Mykhalchuk) wrote…</i></summary>



> Якщо забрати цей пробіл, то виникне помилка, оскільки Robot Framework буде вважати `insert_tender_id_into_xpath ${locator.tender.title.verification}` за ключове слово, а не ключове слово і аргумент.
> </details>
> Done.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4)_

<!-- Sent from Reviewable.io -->

mykhaly commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 
Треба тільки зробити нормальну історію комітів.

---

Reviewed 1 of 1 files at r6.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-:-KNgys0b0jG2jiZ_91Sz:-1101084903)_

<!-- Sent from Reviewable.io -->

mykhaly commented:
<img class="emoji" title=":lgtm_strong:" alt=":lgtm_strong:" align="absmiddle" src="https://reviewable.io/lgtm_strong.png" height="20" width="61"/> 

---

Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-:-KNmBFxsb1G-8dI_vhvO:-1868698381)_

<!-- Sent from Reviewable.io -->

mykhaly commented:
Reviewed 2 of 2 files at r7.
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---

_[publicportal.py, line 6 [r7]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KPIIIg5qMr-d9vXZxrw:-KPIIIg61-0p2mmlQe-6:b-sdiw4b) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/9d22b752d38119a93e1ed7bdf7176cbee86acca6/publicportal.py#L6)):_

> ``` Python
> 
> 
> def parse_date_publicportal(str_date):
> ```

Мені здається, що загальноприйнятішим був би варіант `date_str`, а не `str_date`.

---

_[publicportal.py, line 7 [r7]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KPIIWCGQcenOQCWcKUI:-KPIIWCGQcenOQCWcKUJ:bh0kl5g) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/9d22b752d38119a93e1ed7bdf7176cbee86acca6/publicportal.py#L7)):_

> ``` Python
> 
> def parse_date_publicportal(str_date):
>     split = str_date.split(' ')
> ```

Називати змінну назвою методу якось некласно. Краще вже `foobar`. 
А ще краще отак:

```
def parse_date_publicportal(date_str):
   return ' '.join([date_str.split(' ')[1], date_str.split(' ')[-1]])
```

---

_[publicportal.py, line 12 [r7]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KPIJ6OFZ3S1xMmsiVrs:-KPIJ6OGvv8QK3PAQKI8:b-6yj8lv) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/9d22b752d38119a93e1ed7bdf7176cbee86acca6/publicportal.py#L12)):_

> ``` Python
> 
> 
> def subtract_from_time(date_time, subtr_min, subtr_sec):
> ```

За бажанням можна скоротити за аналогією з функцією вище.

---

_[publicportal.py, line 23 [r7]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KPINnoRGS_DniuSIcM_:-KPINnoRGS_DniuSIcMa:b-2w2399) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/9d22b752d38119a93e1ed7bdf7176cbee86acca6/publicportal.py#L23)):_

> ``` Python
> 
> 
> def adapt_tender_data(tender_data):
> ```

```
def adapt_tender_data(tender_data):
    sum_for_value, sum_for_minimal_step = 0, 0
    for lot in tender_data['data'].get('lots', []):
        lot['value']['amount'] = int(lot['value']['amount'])
        sum_for_value += lot['value']['amount']
        lot['minimalStep']['amount'] = int(lot['minimalStep']['amount'])
        sum_for_minimal_step += lot['minimalStep']['amount']
    tender_data['data']['minimalStep']['amount'] = sum_for_minimal_step if sum_for_minimal_step != 0 else int(tender_data['data']['minimalStep']['amount'])    
    tender_data.data.value['amount'] = sum_for_value if sum_for_value != 0 else int(tender_data.data.value['amount'])
    return tender_data
```

---

_[publicportal.robot, line 36 [r7]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KPIKYZclVV_ev8fOwQj:-KPIKYZde69DhxpoMGcp:b3evsn4) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/9d22b752d38119a93e1ed7bdf7176cbee86acca6/publicportal.robot#L36)):_

> ``` RobotFramework
> 
> Підготувати клієнт для користувача
>   Set Global Variable  ${PUBLICPORTAL_MODIFICATION_DATE}  ${EMPTY}
> ```

Це ніде не використовується.

---

_[publicportal.robot, line 53 [r7]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KPIKq1sxoLdTDWXz0Xd:-KPIKq1sxoLdTDWXz0Xe:b5pnjcx) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/9d22b752d38119a93e1ed7bdf7176cbee86acca6/publicportal.robot#L53)):_

> ``` RobotFramework
> Пошук тендера по ідентифікатору
>   [Arguments]  ${username}  ${tender_uaid}
>   Set Global Variable  ${xpath}  ${EMPTY}
> ```

Навіщо це?

---

_[publicportal.robot, line 103 [r7]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KPILOUZC2YhzhErE7Zh:-KPILOUZC2YhzhErE7Zi:bult2jc) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/9d22b752d38119a93e1ed7bdf7176cbee86acca6/publicportal.robot#L103)):_

> ``` RobotFramework
> 
> 
> Отримати інформацію із предмету
> ```

Таке не підходить. `tender_period.startDate` не завжди рівне `enquiryPeriod.endDate`. Якщо на майданчику не виводиться дата початку періоду пропозицій, то варто так і  пояснювати причину непроходження тесту.

---

_[publicportal.robot, line 116 [r7]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KPIMLMVAh-hRBGUKjZz:-KPIMLMVAh-hRBGUKj_-:br0nerk) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/9d22b752d38119a93e1ed7bdf7176cbee86acca6/publicportal.robot#L116)):_
Це код з підпорками, що є недобре, або опечатка.

---

_[publicportal.robot, line 174 [r7]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KPINKBwhE4VExsbOJ44:-KPINKBxAUhAJX_YZZrW:b-tn2096) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/9d22b752d38119a93e1ed7bdf7176cbee86acca6/publicportal.robot#L174)):_
ВОУ ВОУ1
Тут треба робити `Fail  CAN'T FIND THIS INFO ON PublicPortal`, а не `pass` :)

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4)_

<!-- Sent from Reviewable.io -->

lesiavl commented:
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---

_[publicportal.py, line 1 [r1]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KNaQ6CSph_9WK_QueJ2:-KPIei6b5k7GBtt9RIem:b-896fix) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/d098afc437f976f2e0500b8f02f3ec54e9101026/publicportal.py#L1)):_
<details><summary>

<i>Previously, mykhaly (Yurii Mykhalchuk) wrote…</i></summary>



> :+1: 
> </details>
> Done.

---

_[publicportal.py, line 6 [r7]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KPIIIg5qMr-d9vXZxrw:-KPIef0O5V27_PCVSLIi:b-896fix) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/9d22b752d38119a93e1ed7bdf7176cbee86acca6/publicportal.py#L6)):_
<details><summary>

<i>Previously, mykhaly (Yurii Mykhalchuk) wrote…</i></summary>



> Мені здається, що загальноприйнятішим був би варіант `date_str`, а не `str_date`.
> </details>
> Done.

---

_[publicportal.py, line 7 [r7]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KPIIWCGQcenOQCWcKUI:-KPIee2VLha3tUOKYsvZ:b-896fix) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/9d22b752d38119a93e1ed7bdf7176cbee86acca6/publicportal.py#L7)):_
<details><summary>

<i>Previously, mykhaly (Yurii Mykhalchuk) wrote…</i></summary>



> Називати змінну назвою методу якось некласно. Краще вже `foobar`. 
> А ще краще отак:
> 
> ```
> def parse_date_publicportal(date_str):
>    return ' '.join([date_str.split(' ')[1], date_str.split(' ')[-1]])
> ```
> 
> </details>
> Done.

---

_[publicportal.py, line 12 [r7]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KPIJ6OFZ3S1xMmsiVrs:-KPIfX79fpE6KJNKccLp:b-3qylyz) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/9d22b752d38119a93e1ed7bdf7176cbee86acca6/publicportal.py#L12)):_
<details><summary>

<i>Previously, mykhaly (Yurii Mykhalchuk) wrote…</i></summary>



> За бажанням можна скоротити за аналогією з функцією вище.
> </details>
> Скоротила.

---

_[publicportal.py, line 23 [r7]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KPINnoRGS_DniuSIcM_:-KPIfb3trq9G70VARLUg:b-896fix) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/9d22b752d38119a93e1ed7bdf7176cbee86acca6/publicportal.py#L23)):_
<details><summary>

<i>Previously, mykhaly (Yurii Mykhalchuk) wrote…</i></summary>



> ```
> def adapt_tender_data(tender_data):
>     sum_for_value, sum_for_minimal_step = 0, 0
>     for lot in tender_data['data'].get('lots', []):
>         lot['value']['amount'] = int(lot['value']['amount'])
>         sum_for_value += lot['value']['amount']
>         lot['minimalStep']['amount'] = int(lot['minimalStep']['amount'])
>         sum_for_minimal_step += lot['minimalStep']['amount']
>     tender_data['data']['minimalStep']['amount'] = sum_for_minimal_step if sum_for_minimal_step != 0 else int(tender_data['data']['minimalStep']['amount'])    
>     tender_data.data.value['amount'] = sum_for_value if sum_for_value != 0 else int(tender_data.data.value['amount'])
>     return tender_data
> ```
> 
> </details>
> Done.

---

_[publicportal.robot, line 36 [r7]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KPIKYZclVV_ev8fOwQj:-KPIwHgYOlvjP2xZ1W2Y:b-69xs87) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/9d22b752d38119a93e1ed7bdf7176cbee86acca6/publicportal.robot#L36)):_
<details><summary>

<i>Previously, mykhaly (Yurii Mykhalchuk) wrote…</i></summary>



> Це ніде не використовується.
> </details>
> Видалила

---

_[publicportal.robot, line 53 [r7]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KPIKq1sxoLdTDWXz0Xd:-KPIwEACB3tQpD3SL_hO:b-69xs87) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/9d22b752d38119a93e1ed7bdf7176cbee86acca6/publicportal.robot#L53)):_
<details><summary>

<i>Previously, mykhaly (Yurii Mykhalchuk) wrote…</i></summary>



> Навіщо це?
> </details>
> Видалила

---

_[publicportal.robot, line 116 [r7]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KPIMLMVAh-hRBGUKjZz:-KPIe2meV4FByjman9mW:bnpt78o) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/9d22b752d38119a93e1ed7bdf7176cbee86acca6/publicportal.robot#L116)):_
<details><summary>

<i>Previously, mykhaly (Yurii Mykhalchuk) wrote…</i></summary>



> Це код з підпорками, що є недобре, або опечатка.
> </details>
> замінила

---

_[publicportal.robot, line 174 [r7]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KPINKBwhE4VExsbOJ44:-KPIe-5J0YUWtOn2Yg00:b-896fix) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/9d22b752d38119a93e1ed7bdf7176cbee86acca6/publicportal.robot#L174)):_
<details><summary>

<i>Previously, mykhaly (Yurii Mykhalchuk) wrote…</i></summary>



> ВОУ ВОУ1
> Тут треба робити `Fail  CAN'T FIND THIS INFO ON PublicPortal`, а не `pass` :)
> </details>
> Done.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4)_

<!-- Sent from Reviewable.io -->

lesiavl commented:
Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

_[publicportal.robot, line 103 [r7]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KPILOUZC2YhzhErE7Zh:-KPMv8l8OJhDxRpUvpIA:blb7y1u) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/9d22b752d38119a93e1ed7bdf7176cbee86acca6/publicportal.robot#L103)):_
<details><summary>

<i>Previously, mykhaly (Yurii Mykhalchuk) wrote…</i></summary>



> Таке не підходить. `tender_period.startDate` не завжди рівне `enquiryPeriod.endDate`. Якщо на майданчику не виводиться дата початку періоду пропозицій, то варто так і  пояснювати причину непроходження тесту.
> </details>
> Видалила, дописала причину

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4)_

<!-- Sent from Reviewable.io -->

mykhaly commented:
Reviewed 2 of 2 files at r8.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

_[publicportal.robot, line 155 [r8]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KPMvgD7ynJe_LCGvj6C:-KPMvgD7ynJe_LCGvj6D:b-jy5es3) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/17956c57c19cb4aa1b3204e1220d2e8106e572e2/publicportal.robot#L155)):_
В майбутньому раджу користуватися `Log To Console`.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4)_

<!-- Sent from Reviewable.io -->

lesiavl commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

_[publicportal.robot, line 155 [r8]](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4#-KPMvgD7ynJe_LCGvj6C:-KPMvxoutopBdmCbdufP:b-ck1kjx) ([raw file](https://github.com/openprocurement/robot_tests.broker.publicportal/blob/17956c57c19cb4aa1b3204e1220d2e8106e572e2/publicportal.robot#L155)):_
<details><summary>

<i>Previously, mykhaly (Yurii Mykhalchuk) wrote…</i></summary>



> В майбутньому раджу користуватися `Log To Console`.
> </details>
> Дякую;)

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/openprocurement/robot_tests.broker.publicportal/4)_

<!-- Sent from Reviewable.io -->
